espressostreamer/espresso_streamer.go

```diff
@@ -96,33 +99,62 @@ func NewEspressoStreamer(namespace uint64, hotshotUrls []string,
 		l1Reader:                      l1Reader,
 		espressoTEEVerifierABI:        espressoTEEVerifierAbi,
 		espressoTEEVerifierAddr:       espressoTEEVerifierAddress,
+		// Initally, we set the confirmed message pos to max uint64
+		// because we dont know the confirmed message pos yet
+		confirmedMessagePos: abi.MaxUint256.Uint64(),
 	}
 }
 
-/*
-Pop the first message from the queue
-It will return nil if the queue is empty
-*/
-func (s *EspressoStreamer) PopMessageWithMetadataAndPos() *MessageWithMetadataAndPos {
-	if len(s.messageWithMetadataAndPos) == 0 {
-		return nil
+func (s *EspressoStreamer) Refresh(ctx context.Context, fetchLastMessageAndEspressoBlock func() (uint64, uint64, error)) (bool, error) {
```

Comment:
I am under the impression that this method should just clear the memory queue and reset the `nextHotShotBlock` to a latest valid value.

So here looks strange to me and should espresso streamer does these checks? Probably the caller should check and call this function?

Reply:
So it does reset the queue if it returns true, if it returns false this means there is no need to refresh and it continues as expected. 

The caller is calling the `Refresh` function, you can look at the Caff node code. Its calling the Refresh function

Reply:
New design: https://github.com/EspressoSystems/nitro-espresso-integration/pull/479/commits/760a37441355d2174ac74ecc43b1b3b433d1019c - review this one instead